spring-boot/autoconfigure/src/main/java/org/togglz/spring/boot/autoconfigure/TogglzAutoConfiguration.java

```diff
@@ -0,0 +1,251 @@
+                featureManagerBuilder.name(name);
+            }
+            featureManagerBuilder.featureProvider(featureProvider).stateRepository(stateRepository)
+                    .userProvider(userProvider).activationStrategyProvider(activationStrategyProvider).build();
+            FeatureManager manager = featureManagerBuilder.build();
+            return manager;
+        }
+    }
+
+    @Configuration
+    @ConditionalOnMissingBean(ActivationStrategyProvider.class)
+    protected static class ActivationStrategyProviderConfiguration {
+
+        @Autowired(required = false)
+        private List<ActivationStrategy> activationStrategies;
+
+        @Bean
+        public ActivationStrategyProvider activationStrategyProvider() {
+            DefaultActivationStrategyProvider provider = new DefaultActivationStrategyProvider();
+            if (activationStrategies != null && activationStrategies.size() > 0) {
+                provider.addActivationStrategies(activationStrategies);
+            }
+            return provider;
+        }
+    }
+
+    @Configuration
+    @ConditionalOnMissingBean(StateRepository.class)
+    protected static class StateRepositoryConfiguration {
+
+        @Autowired
+        private ResourceLoader resourceLoader = new DefaultResourceLoader();
+
+        @Autowired
+        private TogglzProperties properties;
+
+        @Bean
+        public StateRepository stateRepository() throws IOException {
+            StateRepository stateRepository;
+            String featuresFile = properties.getFeaturesFile();
+            if (featuresFile != null) {
+                Resource resource = this.resourceLoader.getResource(featuresFile);
+                Integer minCheckInterval = properties.getFeaturesFileMinCheckInterval();
+                if (minCheckInterval != null) {
+                    stateRepository = new FileBasedStateRepository(resource.getFile(), minCheckInterval);
+                } else {
+                    stateRepository = new FileBasedStateRepository(resource.getFile());
+                }
+            } else {
+                Map<String, FeatureSpec> features = properties.getFeatures();
+                stateRepository = new InMemoryStateRepository();
+                for (String name : features.keySet()) {
+                    stateRepository.setFeatureState(features.get(name).state(name));
+                }
+            }
+            // If caching is enabled wrap state repository in caching state
+            // repository.
+            if (properties.getCache().isEnabled()) {
+                stateRepository = new CachingStateRepository(stateRepository, properties.getCache().getTimeToLive(),
+                        properties.getCache().getTimeUnit());
+            }
+            return stateRepository;
+        }
+    }
+
+    @Configuration
+    @ConditionalOnMissingClass("org.springframework.security.config.annotation.web.configuration.EnableWebSecurity")
```

Comment:
Why did you go for a String here?

Reply:
It was already that way, and `@ConditionalOnMissingClass` only supports a string value, not a class.